File: frontend/lib/api/types/career-companion.ts
export type CompanionMode =
  | "target_vacancy"
  | "target_role"
  | "career_growth"
  | "explore_direction";

export type ActionHorizon = "fix_now" | "build_next" | "grow_further";

export type ActionType =
  | "improve_existing_project"
  | "build_new_project"
  | "learn_foundation";

export type ActionStatus =
  | "suggested"
  | "accepted"
  | "in_progress"
  | "awaiting_evidence"
  | "evidence_detected"
  | "partially_verified"
  | "completed"
  | "dismissed";

export type CareerCompanionActionSkill = {
  skill_id: string;
  skill_name: string;
  role: "gap" | "potential_cover" | "related";
};

export type CareerCompanionAction = {
  id: string;
  horizon: ActionHorizon;
  action_type: ActionType;
  status: ActionStatus;
  title: string;
  description: string;
  why_it_matters: string;
  implementation_steps: string[];
  expected_artifacts: string[];
  verification_method: string;
  estimated_effort: string;
  github_repository_id: string | null;
  project_label: string | null;
  current_target_impact: Record<string, unknown>;
  career_growth_impact: Record<string, unknown>;
  priority_score: number;
  priority_explanation: string;
  sort_order: number;
  skills: CareerCompanionActionSkill[];
};

export type CareerCompanionProgressEvent = {
  id: string;
  event_type: string;
  title: string;
  detail: string;
  payload: Record<string, unknown>;
  created_at: string;
};

export type CareerCompanionChatMessage = {
  id: string;
  role: string;
  content: string;
  revision_applied: string | null;
  created_at: string;
};

export type CareerCompanionPlan = {
  id: string;
  mode: CompanionMode;
  target_vacancy_id: string | null;
  target_role: string | null;
  status: string;
  generation_mode: "live" | "mock" | "fallback";
  summary: Record<string, unknown>;
  current_position: {
    goal_label?: string;
    verified_skills?: string[];
    missing_required_skills?: string[];
    missing_preferred_skills?: string[];
    weak_evidence?: string[];
    strongest_projects?: Array<{ id: string; label: string; repository_url: string }>;
    readiness?: string;
    target_match_score?: number | null;
    explore_directions?: string[];
    next_level_titles?: string[];
    has_resume?: boolean;
    mode?: string;
  };
  actions: CareerCompanionAction[];
  progress_events: CareerCompanionProgressEvent[];
  chat_messages: CareerCompanionChatMessage[];
};

export type CareerCompanionGenerateRequest = {
  mode: CompanionMode;
  target_vacancy_id?: string | null;
  target_role?: string | null;
};

export type CareerCompanionChatResponse = {
  message: CareerCompanionChatMessage;
  plan: CareerCompanionPlan | null;
};

File: backend/app/schemas/career_companion.py
from __future__ import annotations

from typing import Any, Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

CompanionMode = Literal[
    "target_vacancy", "target_role", "career_growth", "explore_direction"
]
ActionHorizon = Literal["fix_now", "build_next", "grow_further"]
ActionType = Literal[
    "improve_existing_project", "build_new_project", "learn_foundation"
]
ActionStatus = Literal[
    "suggested",
    "accepted",
    "in_progress",
    "awaiting_evidence",
    "evidence_detected",
    "partially_verified",
    "completed",
    "dismissed",
]
GenerationMode = Literal["live", "mock", "fallback"]
SkillRole = Literal["gap", "potential_cover", "related"]


class CareerCompanionGenerateRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    mode: CompanionMode
    target_vacancy_id: UUID | None = None
    target_role: str | None = Field(default=None, max_length=120)


class CareerCompanionActionStatusPatch(BaseModel):
    model_config = ConfigDict(extra="forbid")

    status: Literal[
        "accepted", "in_progress", "awaiting_evidence", "dismissed"
    ]


class CareerCompanionChatRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    message: str = Field(min_length=1, max_length=2000)


class CareerCompanionActionSkillResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    skill_id: UUID
    skill_name: str
    role: SkillRole


class CareerCompanionActionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    horizon: ActionHorizon
    action_type: ActionType
    status: ActionStatus
    title: str
    description: str
    why_it_matters: str
    implementation_steps: list[str]
    expected_artifacts: list[str]
    verification_method: str
    estimated_effort: str
    github_repository_id: UUID | None
    project_label: str | None
    current_target_impact: dict[str, Any]
    career_growth_impact: dict[str, Any]
    priority_score: float
    priority_explanation: str
    sort_order: int
    skills: list[CareerCompanionActionSkillResponse]


class CareerCompanionProgressEventResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    event_type: str
    title: str
    detail: str
    payload: dict[str, Any]
    created_at: Any


class CareerCompanionChatMessageResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    role: str
    content: str
    revision_applied: str | None
    created_at: Any


class CareerCompanionPlanResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    mode: CompanionMode
    target_vacancy_id: UUID | None
    target_role: str | None
    status: str
    generation_mode: GenerationMode
    summary: dict[str, Any]
    current_position: dict[str, Any]
    actions: list[CareerCompanionActionResponse]
    progress_events: list[CareerCompanionProgressEventResponse]
    chat_messages: list[CareerCompanionChatMessageResponse]


class CareerCompanionChatResponse(BaseModel):
    message: CareerCompanionChatMessageResponse
    plan: CareerCompanionPlanResponse | None = None
